### DISCUSSION TITLE
construct for dataclasses

### DISCUSSION THREAD

**Discussion by espositofulvio** [pydantic/pydantic#2681] [Feature Request]
I'm not sure if there's another way to do this, but it would be useful to have something similar to `BaseModel.construct()` for pydantic dataclassese. In https://github.com/samuelcolvin/pydantic/issues/268 @samuelcolvin pointed out we could switch between standard dataclasses and pydantic dataclasses changing the decorator and getting a validation free dataclass, but I'd like to have something like `construct ` only for when a dataclass object is loaded from a trusted source (a db, for instance) so it's already been validated and thus validation is not needed.

**Comment by PrettyWood** [comment]
Hi @espositofulvio 
Things may change if https://github.com/samuelcolvin/pydantic/pull/2557 is accepted for v1.9, in which case you will be able to bypass validation with `__pydantic_run_validation__=False`
You're welcome to try out the PR to see if it fits your needs

**Reply by espositofulvio** [reply]
Hi @PrettyWood ,
Though I'm confident it has the same effect, I'm just curious if there is a technical reason to go this way instead of adding the construct class method to the dataclass as in BaseModel?.

**Reply by PrettyWood** [reply]
It just depends how much we want to overload `dataclass`.
Btw you can already do what you want by calling the model behind the dataclass
```py
@dataclass
class A:
    x: int

A.__pydantic_model__.construct(x='qweqwe')
```

**Reply by espositofulvio** [reply]
@PrettyWood yes I thought of calling construct that way, I don't know if it's safe to do so or there's something more to do in the dataclass to make it work in all cases. Do you happen to know if something special is required to make it work?

**Reply by PrettyWood** [reply]
nothing should be required
